src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -991,27 +991,59 @@ private static void cachePut(Tangle tangle, TransactionViewModel transactionView
      */
     public static void cacheEvict(Tangle tangle) throws Exception {
         Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
+        List<Pair<Indexable, Persistable>> batch = new ArrayList<>();
         for (int i = 0; i < cache.getConfiguration().getEvictionCount(); i++) {
             Indexable hash = cache.nextEvictionKey();
             if (hash != null) {
                 TransactionViewModel tvm = cache.get(hash);
                 if (tvm != null) {
                     if (!tvm.getIsCacheEntryFresh()) {
-                        tvm.updateDB(tangle, tvm.getTransaction(), tvm.getHash(), "");
+                        batch.addAll(tvm.getSaveBatch());
+                        cache.release(tvm.getHash());
                     }
-                    cache.evict(tvm.getHash());
                 }
             }
         }
+        tangle.saveBatch(batch);
```

Comment:
I think we should save before releasing from cache
Imagine that the save fails...

Reply:
My comment above still holds, and I think there may be some release txs we forget to save

Reply:
Currently, I don't see how this could happen.
If a tx is `fresh`, then it's already in DB.

Maybe the culprit here is the term `fresh` that stems from the original requirement that envisaged a cleaner design.
I think `isDirty` will be clearer to understand.
A tx is `dirty` if it's been 
1. Cached
2. Updated and cached again

So only `dirty` txs will be persisted upon cache release.